pkgs/os-specific/linux/fuse/common.nix

```diff
@@ -43,34 +43,34 @@ in stdenv.mkDerivation rec {
     then [ meson ninja pkg-config ]
     else [ autoreconfHook gettext ];
 
-  outputs = [ "out" ] ++ lib.optional isFuse3 "common";
+  outputs = [ "bin" "out" "dev" "man" ] ++ lib.optional isFuse3 "common";
 
   mesonFlags = lib.optionals isFuse3 [
     "-Dudevrulesdir=/udev/rules.d"
     "-Duseroot=false"
+    # `fuserconf_path` is pinned to `/etc/fuse.conf` in `./fuse3-install.patch`,
+    # we are free to modify this option now.
+    "-Dsysconfdir=${placeholder "common"}/etc"
   ];
 
   preConfigure = ''
-    export MOUNT_FUSE_PATH=$out/sbin
-    export INIT_D_PATH=$TMPDIR/etc/init.d
-    export UDEV_RULES_PATH=$out/etc/udev/rules.d
-
     # Ensure that FUSE calls the setuid wrapper, not
     # $out/bin/fusermount. It falls back to calling fusermount in
     # $PATH, so it should also work on non-NixOS systems.
     export NIX_CFLAGS_COMPILE="-DFUSERMOUNT_DIR=\"/run/wrappers/bin\""
 
     substituteInPlace lib/mount_util.c --replace "/bin/" "${util-linux}/bin/"
-    '' + (if isFuse3 then ''
-      # The configure phase will delete these files (temporary workaround for
-      # ./fuse3-install_man.patch)
-      install -D -m444 doc/fusermount3.1 $out/share/man/man1/fusermount3.1
-      install -D -m444 doc/mount.fuse3.8 $out/share/man/man8/mount.fuse3.8
-    '' else ''
-      substituteInPlace util/mount.fuse.c --replace '"su"' '"${shadow.su}/bin/su"'
-      sed -e 's@CONFIG_RPATH=/usr/share/gettext/config.rpath@CONFIG_RPATH=${gettext}/share/gettext/config.rpath@' -i makeconf.sh
-      ./makeconf.sh
-    '');
+  '' + lib.optionalString (!isFuse3) ''
+    export MOUNT_FUSE_PATH=$bin/bin
+
+    # Ignored for fuse2, see preFixup.
+    export INIT_D_PATH=$TMPDIR/etc/init.d
+    export UDEV_RULES_PATH=$TMPDIR/etc/udev/rules.d
+
+    substituteInPlace util/mount.fuse.c --replace '"su"' '"${shadow.su}/bin/su"'
```

Comment:
that will not work because su has no suid bit.

Reply:
This line already exists before this PR.
`su` is used [here](https://github.com/libfuse/libfuse/blob/fuse-2.9.9/util/mount.fuse.c#L216) in `mount.fuse` for some reason. I'd expect user to call this binary with `sudo`, yes? Maybe we should just patch this code out.

Reply:
:thinking: Wait, so this `su` is never working before. I'm a little confused.

Reply:
the line was added in https://github.com/NixOS/nixpkgs/pull/119562

Reply:
su -c will never work for an unprivileged user, so using non-setuid su is ok.
Tthis code path is taken when the option setuid= is specified. man mount.fuse3 says:
>     setuid=USER
              Switch to USER and its primary group before launching the FUSE file system process.
              mount.fuse3 must be run as root or with CAP_SETUID and CAP_SETGID for this to work.